enumerables.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all?(*array)
+    array = nil
+    if array
+      my_each { |x| return false unless array == x }
+    elsif !block_given?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless yield(x) }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    args = nil
+    if args
+      my_each { |x| return true if x == args }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if yield(x) }
+    end
+    false
+  end
+
+  def my_none?(*args, &block)
+    !my_any?
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] returns true if the block never returns true for all elements
```ruby
[0,5,4,9].my_none? {|num| num > 100} == [0,5,4,9].none? {|num| num > 100} => true
```
- Yours is retrieving thew wrong answer
![image](https://user-images.githubusercontent.com/52209067/80133480-84d00e80-855a-11ea-8de5-d73b5a2ce549.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/80133513-931e2a80-855a-11ea-8a3e-3c8bff2758fd.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 9].my_none?(String) == [1, 2.5, 9].none?(String) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80141759-4e4cc080-8567-11ea-8df8-750c73975fb2.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/80141824-66244480-8567-11ea-8a7a-88c6e16e3d0d.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80141905-8bb14e00-8567-11ea-8dcc-b815b43d207e.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[4,7,11].my_none?(3) == [4,7,11].none?(3) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80142001-b6030b80-8567-11ea-8ce4-fb30c19937d2.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/80142038-c2876400-8567-11ea-9886-244bf7629228.png)